features/support/time_machine_helpers.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module TimeMachineHelpers
+  def time_machine(till)
+    travel_to(Time.zone.now)
+
+    while Time.zone.now < till
+      travel_to(1.day.from_now)
+      run_jobs
+    end
+    travel_to(till)
+  rescue StandardError
+    travel_back
+    raise
+  end
+
+  private
+
+  def run_jobs
+    run(ThreeScale::Jobs::MONTH) if Time.zone.now == Time.zone.now.beginning_of_month
+    run(ThreeScale::Jobs::WEEK) if Time.zone.now == Time.zone.now.beginning_of_week
+    run(ThreeScale::Jobs::DAILY)
+    run(ThreeScale::Jobs::BILLING)
```

Comment:
From my understanding, if we add another constant to `ThreeScale::Jobs`, we'll have to also add it here. Is that correct?

Reply:
Reek or Rubocop (never remember which) should be complaining about calling `Time.zone` multiple times. Maybe even `Time.zone.now`. If we intend to keep this as it is, we should add the magic comments so it's clear we didn't agree with the warnings.

Reply:
> From my understanding, if we add another constant to ThreeScale::Jobs, we'll have to also add it here. Is that correct?

I think it would be a good idea to do so, since we want to simulate the pass of time in the real world. So if we added a new list of frequent/scheduled jobs we would want to simulate that to check whether that breaks any test.

Reply:
> Reek or Rubocop (never remember which) should be complaining about calling Time.zone multiple times. Maybe even Time.zone.now. If we intend to keep this as it is, we should add the magic comments so it's clear we didn't agree with the warnings.

Yes I have codeclimate complaining about that but I ignored it so far because it makes no sense to complain about calling `Time.now` repeatedly when it returns a different value every time you call it. This particular case could be an exception because `Time.now` is frozen here.